nifi-nar-bundles/nifi-adx-bundle/nifi-adx-processors/src/main/java/org/apache/nifi/processors/adx/AzureAdxSourceProcessor.java

```diff
@@ -0,0 +1,216 @@
+            .description(AzureAdxSourceProcessorParamsEnum.ADX_QUERY.getParamDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor ADX_SOURCE_SERVICE = new PropertyDescriptor
+            .Builder().name(AzureAdxSourceProcessorParamsEnum.ADX_SOURCE_SERVICE.name())
+            .displayName(AzureAdxSourceProcessorParamsEnum.ADX_SOURCE_SERVICE.getParamDisplayName())
+            .description(AzureAdxSourceProcessorParamsEnum.ADX_SOURCE_SERVICE.getParamDescription())
+            .required(true)
+            .identifiesControllerService(AdxSourceConnectionService.class)
+            .build();
+    private AdxSourceConnectionService service;
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> descriptors;
+    private Client executionClient;
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(ADX_SOURCE_SERVICE);
+        descriptors.add(DB_NAME);
+        descriptors.add(ADX_QUERY);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(RL_SUCCEEDED);
+        relationships.add(RL_FAILED);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        service = context.getProperty(ADX_SOURCE_SERVICE).asControllerService(AdxSourceConnectionService.class);
+        executionClient = service.getKustoExecutionClient();
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        FlowFile outgoingFlowFile;
+
+        String databaseName = context.getProperty(DB_NAME).getValue();
+        String adxQuery = null;
+        KustoResultSetTable kustoResultSetTable;
+
+        if ( context.hasIncomingConnection() ) {
+            FlowFile incomingFlowFile = session.get();
+
+            if ( incomingFlowFile == null && context.hasNonLoopConnection() ) {
+                String message = "Incoming connection exists but Incoming flow file is null";
+                getLogger().error(message);
+                incomingFlowFile = session.putAttribute(incomingFlowFile, ADX_QUERY_ERROR_MESSAGE, message);
+                session.transfer(incomingFlowFile, RL_FAILED);
+                return;
+            }
+
+            if ( incomingFlowFile != null && incomingFlowFile.getSize() == 0 ) {
```

Comment:
Missing comments. Please annotate , makes it very hard to understand what these conditions really mean